pkg/kubelet/kubelet.go

```diff
@@ -1610,11 +1610,7 @@ func (kl *Kubelet) syncPod(o syncPodOptions) error {
 	}
 
 	// Wait for volumes to attach/mount
-	defaultedPod, _, err := kl.defaultPodLimitsForDownwardApi(pod, nil)
```

Comment:
I think the issue is not with this method. The issue is with passing a modified copy of pods to volumeManager.WaitForAttachAndMount which does not get trickled down to volumes. So the changes to the pod are not reflected in volumes. WaitForAttachAndMount only takes name of pods and maintains it own cache of pods.